### PR TITLE
Fix fragment id crash

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by mrc0mmand** [comment]
All CIs now fail in `TEST-15-DROPIN`, seems related:

```
[  422.453997] testsuite-15.sh[30]: + systemctl daemon-reload
[  422.465347] systemd[1]: Reloading.
[  422.708948] testsuite-15.sh[30]: + cat
[  422.710832] testsuite-15.sh[30]: + mkdir -p /etc/systemd/system/test15-b.service.d /run/systemd/system/test15-b.service.d /usr/lib/systemd/system/test15-b.service.d
[  422.712494] testsuite-15.sh[30]: + mkdir -p /etc/systemd/system/test15-b.service.wants /etc/systemd/system/test15-b.service.requires
[  422.714005] testsuite-15.sh[30]: + mkdir -p /run/systemd/system/test15-b.service.wants /run/systemd/system/test15-b.service.requires
[  422.715438] testsuite-15.sh[30]: + mkdir -p /usr/lib/systemd/system/test15-b.service.wants /usr/lib/systemd/system/test15-b.service.requires
[  422.716965] testsuite-15.sh[30]: + ln -sf test15-b.service /etc/systemd/system/test15-b1.service
[  422.718391] testsuite-15.sh[30]: + ln -sf ../test15-b1.service /etc/systemd/system/test15-a.service.wants/
[  422.719811] testsuite-15.sh[30]: + check_ok test15-a Wants test15-b.service
[  422.719811] testsuite-15.sh[30]: + '[' 3 -eq 3 ']'
[  422.720542] testsuite-15.sh[902]: ++ systemctl show --value -p Wants test15-a
[  422.754414] testsuite-15.sh[30]: + x=test15-b1.service
[  422.754414] testsuite-15.sh[30]: + case "$x" in
[  422.754414] testsuite-15.sh[30]: + return 1
[  422.755987] systemd[1]: testsuite-15.service: Main process exited, code=exited, status=1/FAILURE
[  422.756244] systemd[1]: testsuite-15.service: Failed with result 'exit-code'.
[  422.757045] systemd[1]: Failed to start TEST-15-DROPIN.
```

**Comment by keszybz** [comment]
@mrc0mmand thanks. I misunderstood this code completely. I added some comments and a test now.

**Review by bluca — APPROVED** [review]
Nice catch! LGTM

File: test/units/testsuite-15.sh [hunk 1]
```diff
@@ -124,6 +124,32 @@ EOF
...
+
+    echo "*** test linked unit (cross basename)"
+
+    mv /test15-a.service /test15-a@.scope
+    ln -fs /test15-a@.scope /etc/systemd/system/test15-a.service
+    systemctl daemon-reload
+
+    check_ok test15-a Names test15-a.service
+    check_ok test15-a Names test15-b.service
+    check_ko test15-a Names test15-b@
```
Comment (anitazha) [inline test/units/testsuite-15.sh:147]: Was this supposed to be `test15-a@` and not `test15-b@`? I'm puzzled as to why `test15-b@` would have been a possible option

Comment (keszybz) [reply test/units/testsuite-15.sh:147]: I'm not sure any more what I had in mind ;) But yeah, `test15-@` would make more sense.

Comment (keszybz) [reply test/units/testsuite-15.sh:147]: https://github.com/systemd/systemd/pull/18911

**Pull request merged**